daffodil-test-ibm1/src/test/scala/org/apache/daffodil/TresysTests.scala

```diff
@@ -34,69 +33,72 @@ object TresysTests {
-  val av2 = testDir + "AV002.tdml"
-  lazy val runnerAV002 = new DFDLTestSuite(Misc.getRequiredResource(av2))
-  val av3 = testDir + "AV003.tdml"
-  lazy val runnerAV003 = new DFDLTestSuite(Misc.getRequiredResource(av3))
-  val nsd = testDir + "nested-separator-delimited.tdml"
-  lazy val runnerNSD = new DFDLTestSuite(Misc.getRequiredResource(nsd))
-
-
-  lazy val runnerRD = Runner(testDir, "runtime-diagnostics.tdml", compileAllTopLevel = true, validateTDMLFile = false)
-
-  val sq = testDir + "sequence.tdml"
-  lazy val runnerSQ = new DFDLTestSuite(Misc.getRequiredResource(sq))
-
-  lazy val runnerNG = new DFDLTestSuite(Misc.getRequiredResource(testDir + "nested_group_ref.tdml"))
-  val af = testDir + "AF.tdml"
-  lazy val runnerAF = new DFDLTestSuite(Misc.getRequiredResource(af))
-  val ag = testDir + "AG.tdml"
-  lazy val runnerAG = new DFDLTestSuite(Misc.getRequiredResource(ag))
-
-  val aw = testDir + "AW.tdml"
-  lazy val runnerAW = new DFDLTestSuite(Misc.getRequiredResource(aw))
-
-  val ay = testDir + "AY.tdml"
-  lazy val runnerAY = new DFDLTestSuite(Misc.getRequiredResource(ay))
-
-  val az = testDir + "AZ.tdml"
-  lazy val runnerAZ = new DFDLTestSuite(Misc.getRequiredResource(az))
-
-  val ba = testDir + "BA.tdml"
-  lazy val runnerBA = new DFDLTestSuite(Misc.getRequiredResource(ba))
-
-  val bc = testDir + "BC.tdml"
-  lazy val runnerBC = new DFDLTestSuite(Misc.getRequiredResource(bc))
-  val bd = testDir + "BD.tdml"
-  lazy val runnerBD = new DFDLTestSuite(Misc.getRequiredResource(bd))
+  lazy val runnerBB = Runner(testDir, "BB.tdml")
+  lazy val runnerBE = Runner(testDir, "BE.tdml")
+  lazy val runnerBF1 = Runner(testDir, "BF.tdml")
+  lazy val runnerBG = Runner(testDir, "BG.tdml")
+  lazy val runnerMB = Runner(testDir, "mixed-binary-text.tdml")
+
+  lazy val runnerAP = Runner(testDir, "AP.tdml")
+  lazy val runnerAX = Runner(testDir, "AX.tdml")
+  lazy val runnerAV000 = Runner(testDir, "AV000.tdml")
+  lazy val runnerAV001 = Runner(testDir, "AV001.tdml")
+  lazy val runnerAV002 = Runner(testDir, "AV002.tdml")
+  lazy val runnerAV003 = Runner(testDir, "AV003.tdml")
+  lazy val runnerNSD = Runner(testDir, "nested-separator-delimited.tdml")
+
+  lazy val runnerRD = Runner(testDir, "runtime-diagnostics.tdml", RunnerOpts(compileAllTopLevel = true, validateTDMLFile = false))
+
+  lazy val runnerSQ = Runner(testDir, "sequence.tdml")
+
+  lazy val runnerNG = Runner(testDir, "nested_group_ref.tdml")
+
+  lazy val runnerAF = Runner(testDir, "AF.tdml")
+
+  lazy val runnerAG = Runner(testDir, "AG.tdml")
+
+  lazy val runnerAW = Runner(testDir, "AW.tdml")
+
+  lazy val runnerAY = Runner(testDir, "AY.tdml")
+
+  lazy val runnerAZ = Runner(testDir, "AZ.tdml")
+
+  lazy val runnerBA = Runner(testDir, "BA.tdml")
+
+  lazy val runnerBC = Runner(testDir, "BC.tdml")
+
+  lazy val runnerBD = Runner(testDir, "BD.tdml")
 
   @AfterClass def shutDown: Unit = {
     runnerDelimited.reset
     runnerMD.reset
     runnerMD_NV.reset
+    runnerBB.reset
```

Comment:
Since we are modifying so many test files do we also want to add the ``shutDown`` method to reset the runners? I think many of them are missing.

I'm wondering if a convenient way to do this would be to add a new mixin (i.e. DaffodilTestSuite) that implements shutDown and uses reflection to find any Runner members and automatically call reset on them?

Reply:
I would call this mixin AllRunnersAutoShutdownMixin.